projects/utils-maps/src/lib/ol/utils-ol-mapstate.ts

```diff
@@ -0,0 +1,204 @@
+  const srcProjection = getProjection(map.getView().getProjection().getCode());
+  const transformedCenter = transform(map.getView().getCenter(), srcProjection, dstProjection);
+  return transformedCenter;
+}
+
+
+/**
+  *
+  * @param features: olFeature[]
+  * @param geographic: boolean
+  * @returns olExtend: [minX, minY, maxX, maxY]
+  */
+export function getFeaturesExtent(map: Map, features: Feature<any>[], geographic?: boolean) {
+  const extent: any = features[0].getGeometry().getExtent().slice(0);
+  features.forEach((feature) => {
+    olExtend(extent, feature.getGeometry().getExtent());
+  });
+  if (geographic) {
+    const projection = getProjection('EPSG:4326');
+    const transformedExtent = transformExtent(extent, map.getView().getProjection().getCode(), projection);
+    return (transformedExtent as Extent);
+  } else {
+    return extent;
+  }
+}
+
+/**
+ * @param geographic: boolean
+ * @returns olExtend: [minX, minY, maxX, maxY]
+ */
+export function getCurrentExtent(map: Map, geographic?: boolean) {
+  const projection = (geographic) ? getProjection('EPSG:4326') : getProjection(DEFAULT_PROJECTION);
+  const extent = map.getView().calculateExtent();
+  const transformedExtent = transformExtent(extent, map.getView().getProjection().getCode(), projection);
+  return (transformedExtent as Extent);
+}
+
+
+export function setZoom(map: Map, zoom: number, notifier?: 'map' | 'user') {
+  const view = map.getView();
+  view.setZoom(zoom);
+}
+
+/** USED in map-ol.component */
+export function getZoom(map: Map): number {
+  return map.getView().getZoom();
+}
+
+export function zoomInOut(map: Map, value: '-' | '+') {
+  const view = map.getView();
+  if (!view) {
+    // the map does not have a view, so we can't act
+    // upon it
+    return;
+  }
+  const delta = 1, duration = 250;
+  const currentZoom = view.getZoom();
+  if (currentZoom !== undefined) {
+    const newZoom = view.getConstrainedZoom(currentZoom + delta);
+    if (duration > 0) {
+      if (view.getAnimating()) {
+        view.cancelAnimations();
+      }
+      view.animate({
+        zoom: newZoom,
+        duration,
+        easing: easeOut
+      });
+    } else {
+      view.setZoom(newZoom);
+    }
+  }
+}
+
+
+
+export function getMapProjection(map: Map) {
+  return map.getView().getProjection();
+}
+
+
+/**
+ * vector layers will be reprojected automatically
+ * wms layers will be updated with corresponding proj def in the requests.
+ * for other raster layers and for those wms layers whose backend does not support target projection, please
+ * define initial(default) layer projection, so openlayers will reproject on the client side
+ * projection is proj~ProjectionLike
+ */
+export function setMapProjection(map: Map, projection: Projection | string, viewOptions?: ViewOptions, cb?: (projection: Projection) => void) {
+  let epsg = DEFAULT_PROJECTION;
+  let view = null;
+  if (projection) {
```

Comment:
This cannot be false, except is the caller provides an empty string. Do we really want to test for that or should I remove this line?

Reply:
I'm not the only person with implemented this function. 
maybe we could ask @voinSR for this.
But i think you're right :)

Reply:
i think you both correct :) I don't know anymore why I did this :)

Reply:
Great! Will remove the check. Thanks, guys!